### PR TITLE
GRAD2-2952: task is complete.

### PR DESCRIPTION
**Pull request by infstar**
GRAD2-2952: task is complete.

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/distribution/model/dto/v2/School.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	private String schoolCategoryCode;
+	private String schoolCategoryCodeInstitute;
+
+	@JsonIgnore
+	private String schlNo;
+	@JsonIgnore
+	private String distNo;
+    
+	public String getSchoolName() {
+		return  schoolName != null ? schoolName.trim(): null;
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/distribution/model/dto/v2/School.java:33]: Is the trim necessary? Could just use lombok Getter

File: api/src/main/java/ca/bc/gov/educ/api/distribution/process/BaseProcess.java [hunk 2]
```diff
@@ -169,12 +170,12 @@ protected void processSchoolsForLabels(List<School> schools, Psi psi) {
...
-    protected void processSchoolsForLabels(String recipient, List<School> schools, String mincode, ExceptionMessage exception) {
-        School existSchool = schools.stream().filter(s -> mincode.equalsIgnoreCase(s.getMincode())).findAny().orElse(null);
+    protected void processSchoolsForLabels(String recipient, List<ca.bc.gov.educ.api.distribution.model.dto.School> schools, String mincode, ExceptionMessage exception) {
+        ca.bc.gov.educ.api.distribution.model.dto.School existSchool = schools.stream().filter(s -> mincode.equalsIgnoreCase(s.getMincode())).findAny().orElse(null);
         if (existSchool != null) return;
-        TraxSchool traxSchool = schoolService.getTraxSchool(mincode, exception);
+        ca.bc.gov.educ.api.distribution.model.dto.v2.School traxSchool = schoolService.getSchool(mincode, exception);
         if (traxSchool != null) {
-            School school = new School();
+            ca.bc.gov.educ.api.distribution.model.dto.School school = new ca.bc.gov.educ.api.distribution.model.dto.School();
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/distribution/process/BaseProcess.java:178]: Should these all be v2 schools?

File: api/src/main/java/ca/bc/gov/educ/api/distribution/process/MergeProcess.java [hunk 3]
```diff
@@ -49,12 +49,12 @@ public ProcessorData fire(ProcessorData processorData) {
 		Long batchId = processorData.getBatchId();
 		int numberOfPdfs = 0;
 		int counter = 0;
-		List<School> schoolsForLabels = new ArrayList<>();
+		List<ca.bc.gov.educ.api.distribution.model.dto.School> schoolsForLabels = new ArrayList<>();
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/distribution/process/MergeProcess.java:52]: should be v2?

Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/distribution/process/MergeProcess.java:52]: This is right as School DTO here is for Report Data being used by Grad-Report-Api.  Some APIs have it's name as SchoolReport.java.

File: api/src/main/java/ca/bc/gov/educ/api/distribution/process/BaseProcess.java (hunk 2 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/distribution/process/BaseProcess.java:178]: This is right as School DTO here is for Report Data being used by Grad-Report-Api.

File: api/src/main/java/ca/bc/gov/educ/api/distribution/model/dto/v2/School.java (hunk 1 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/distribution/model/dto/v2/School.java:33]: Good catch.  I think it is copied from TraxSchool as it's open vms database fills space if the actual value is less than the field size defined in the table.  I will update it soon.

**Review by mightycox — APPROVED** [review]

**Pull request merged**